changelog.d/20230830_153211_30907815+rjmello_self_diagnostic_env.rst

```diff
@@ -0,0 +1,7 @@
+New Functionality
+^^^^^^^^^^^^^^^^^
+
+- The `self-diagnostic` endpoint command is now aware of the `GLOBUS_SDK_ENVIRONMENT`
+  environment variable and will test connectivity to the relevant services.
```

Comment:
I suggest not mentioning this in the changelog.  The `GLOBUS_SDK_ENVIRONMENT` is a client-side variable for directing requests to one of our pre-production environments (which nominally no end-user will ever access).  Meaning, that it's a useful tidbit of which to be aware for the developers ("us"), but I'd argue is along the lines of TMI for almost all users.